### ISSUE TITLE
Setting metadata failure

### ISSUE THREAD

**Issue by bernt-matthias** [galaxyproject/galaxy#15568]
**Describe the bug**

```
setting metadata externally failed for HistoryDatasetAssociation 1471087: Metadata results could not be read from '/work/songalax/galaxy-dev/database/jobs_directory/021/21650/metadata/metadata_results___new_primary_file_genome_fasta|GCF_016746235.2:GCF_016746235.2_Prin_Dtei_1.1__'
```

Maybe due to the `|` in the filename?

**Galaxy Version and/or server at which you observed the bug**
Galaxy Version: 22.05 
Commit: 5fd1141c385767ba81e2eaf2f299adf81b494fa3

**To Reproduce**

I downloaded some drosophila genomes with the IUC's NCBI  datasets tool.

**Expected behavior**

No error

**Screenshots**

**Additional context**

**Comment by mvdbeek** [comment]
well, that can have a lot of causes. you'll want to check if there's a traceback object attached in the job info page, and whether you can run the metdata script manually. `|` should not be a problem.

**Comment by bernt-matthias** [comment]
Running 

```
GALAXY_LIB="/gpfs1/data/galaxy_server/galaxy-dev/lib"
PYTHONPATH="$GALAXY_LIB"
. /gpfs1/data/galaxy_server/galaxy-dev/.venv/bin/activate
python "metadata/set.py"
```

worked ... 

> Traceback object

Is this already available on 22.05 .. I could find nothing

**Comment by mvdbeek** [comment]
yes, it's in the job info page iff there is a traceback attached to your job.
run the job again ? Could be a problem with the real user setup too ?
